### PR TITLE
Add validation function and function to compute top-1 and top-5 error…

### PR DESCRIPTION
**Pull request by aimeouraga**
- Implemented the `top_k_error_rate` function to compute the top-1 and top-5 error rates for model evaluation.
- Added the `validate` function to evaluate the model on the validation set, including loss, accuracy, and error rates.
- Configured device setup to use CUDA if available.

### CODE REVIEW

File: utils/evaluate.py [hunk 1]
```diff
@@ -2,10 +2,62 @@
 
 
 # Device configuration
-
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 
 # Define top_k_error_rate
+def top_k_error_rate(output, target, k=5):
+    """Given the output and the target, we compute the top-1 and top-5 error rates."""
```
Comment (Omer-alt) [inline utils/evaluate.py:9]: Good practice commenting on functions.

File: utils/evaluate.py [hunk 2]
```diff
@@ -2,10 +2,62 @@
...
+            loss = criterion(outputs, labels)
+            val_loss += loss.item()
+            total_images += labels.size(0)
+            
+            _, predicted = torch.max(outputs.data, 1)
+            total_val += labels.size(0)
+            correct_val += (predicted == labels).sum().item()
+
+            # Compute top-1 and top-5 error rates
+            top1_error_rate, top5_error_rate = top_k_error_rate(outputs, labels, k=5)[:2]
```
Comment (Omer-alt) [inline utils/evaluate.py:52]: How can we be sure that the result of this function is necessarily top1 and top5 error rate ?
